game-core/src/main/java/games/strategy/engine/data/GameData.java

```diff
@@ -143,27 +143,10 @@ public GameMap getMap() {
     return map;
   }
 
-  /**
-   * Print an exception report if we are testing the lock is held, and
-   * do not currently hold the read or write lock.
-   */
-  private void ensureLockHeld() {
-    if (!testLockIsHeld) {
-      return;
-    }
-    if (readWriteLockMissing()) {
-      return;
-    }
-    if (!lockUtil.isLockHeld(readWriteLock.readLock()) && !lockUtil.isLockHeld(readWriteLock.writeLock())) {
-      new Exception("Lock not held").printStackTrace(System.out);
```

Comment:
This was interesting to see, we print out a stack trace here and would not print out the exception message. This would be pretty hard to debug as it'll be mysterious what the stack trace was printed for. Also, the trace was going to std out, so it was 'quiet'. Unless something is logged at a finer level, the 'quiet' logging in general is all mistaken error handling. I've been iterating over the error handling for some time, introducing ClientLogger, I noted that much error handling was a bit timid and we had multiple ways of doing it. The usage of 'quiet' logging is an example. Considering the code has run for so long, we've had no reports with the stack trace from this, I would conclude this to be effectively debug/dead code; hence the removal.